### ISSUE TITLE
moving caret will only select text after making a selection in visual mode

### ISSUE THREAD

**Issue by LaptopDev** [gdh1995/vimium-c#1160]
**Describe the bug**
cannot move caret without creating selection in visual mode. Seems that it will not de-select text in visual mode or does not re-set itself in some way as to fully escape selection, even after exiting visual mode and re-entering visual mode

moving caret will only select text after making a selection in visual mode

**To Reproduce**
1. in visual mode, press v to select text
2. select text
3. exit visual mode
4. clear selection by pressing escape
5. re-enter visual mode
6. move caret
> >moving caret creates text selection trail< <

**Browser, OS and Vimium C versions**

- Browser name: Firefox
- Browser version: firefox.x86_64 127.0.2-1.fc40
- Vimium C version: Vimium C [1.99.997]
- OS name and version: Fedora Linux 40

**Comment by gdh1995** [comment]
you may try "caret mode", which can be entered by pressing `c` if you have been in visual mode.

`map xxx enterInsertMode mode="caret"` may also help.

**Comment by LaptopDev** [comment]
yes, I return to normal vi motions by pressing c in visual mode with an active selection to stop selection and move caret

glad there is currently a work around, thank you.

**Comment by gdh1995** [comment]
Sorry I can not get what you want yet.

The `caret mode` is a feature, but not a working around. And a selection with a width of at least one character is necessary for users to know where a current caret is easily.

**Comment by LaptopDev** [comment]
Forgive me. I am only using the extension and unfamiliar with the code.

What file contains the logic for selections? I don't understand why or how selection width could be causing the behavior.

**Comment by gdh1995** [comment]
Sorry again. In fact I'm not good at the text editor software VIM, so I didn't get the meaning of `moving caret creates text selection trail< <` in the issue description yet.

Could you give some more explanations with simple words suitable for non-programmer browser users? Maybe some snapshots add red circles on them could help.

**Comment by groutoutlook** [comment]
> you may try "caret mode", which can be entered by pressing `c` if you have been in visual mode.
> 
> `map xxx enterInsertMode mode="caret"` may also help.

FYI it's `map c enterVisualMode mode="caret"` 

My use case is very wicked, I turn on caret browsing on msedge by default. Took me 20 minutes to pull this out.
